src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -388,26 +423,30 @@ FirebaseService getService(String id) {
    */
   static class TokenRefresher implements CredentialsChangedListener {
 
+    private static final int STATE_READY = 0;
```

Comment:
Can we use an Enum here and an AtomicReference below?

It may even be fine to just use a volatile enum: http://www.javamex.com/tutorials/synchronization_volatile.shtml

Reply:
Done. Used enum. Kept the atomic reference for compare-and-set method.